code/go/0chain.net/chaincore/chain/entity.go

```diff
@@ -1367,6 +1370,68 @@ func (c *Chain) SetLatestFinalizedBlock(b *block.Block) {
 	}
 }
 
+func (mc *Chain) getClientState(pb *block.Block) (util.MerklePatriciaTrieI, error) {
+	if pb == nil || pb.ClientState == nil {
+		return nil, fmt.Errorf("cannot get MPT from latest finalized block %v", pb)
+	}
+	return pb.ClientState, nil
+}
+
+func getConfigMap(clientState util.MerklePatriciaTrieI) (*minersc.GlobalSettings, error) {
+	if clientState == nil {
+		return nil, errors.New("client state is nil")
+	}
+
+	val, err := clientState.GetNodeValue(util.Path(encryption.Hash(minersc.GLOBALS_KEY)))
+	if err != nil {
+		return nil, err
+	}
+
+	gl := &minersc.GlobalSettings{
+		Fields: make(map[string]string),
+	}
+	err = gl.Decode(val.Encode())
+	if err != nil {
+		return nil, err
+	}
+
+	return gl, nil
+}
+
+func (mc *Chain) updateConfig(pb *block.Block) {
+	clientState, err := mc.getClientState(pb)
+	if err != nil {
+		// This might happen after stopping and starting the miners
+		// and the MPT has not been setup yet.
```

Comment:
Does this handle https://github.com/0chain/0chain/issues/541, if so can we link the issue to this Pr?